src/util/text_parser.ts

```diff
@@ -76,3 +77,27 @@ export const extractIssuesOrPullRequestMarkdownLinks = (body: string) => {
 
   return results;
 };
+
+export const extractTasks = (body: string) => {
+  const matchAll = /- \[( |)(x|X| |)(| )\] /;
+  const matchChecked = /- \[( |)(x|X)(| )\] /;
+  let tasks: PullOrBodyTask[] = [];
+
+  body.split("\n").forEach((line: string) => {
+    let checked: boolean = false;
+    let description: string;
+    if (!line.trim().startsWith("- [")) {
+      return;
+    }
+
+    if (matchChecked.test(line)) {
+      checked = true;
+    }
+    description = line
+      .split(matchAll)
+      [line.split(matchAll).length - 1].trim()
```

Comment:
You're coding yourself into a corner by trying to do it in 1 expression. You end up splitting the line twice. Better to break it up.